examples/ex3p.cpp

```diff
@@ -26,6 +26,10 @@
 //               mpirun -np 4 ex3p -m ../data/star.mesh -pa -d raja-omp
 //               mpirun -np 4 ex3p -m ../data/beam-hex.mesh -pa -d cuda
 //
+// Device w/o partial assembly runs:
```

Comment:
These sample runs are helpful during the review, but maybe they should not be included in the master branch. The first one, `-no-pa -d cpu`, is the same as the first sample run above (line 5). The second one, `-no-pa -d cuda`, could be kept but probably does not have significant speed-up compared with `-d cpu` since the hypre solve is still on CPU. The second run is what this PR fixes (in terms of correctness, not speed), and it is good to have it fixed, but I'm not sure there is value in highlighting it as a sample run. I would be fine with keeping it or not.

Reply:
I agree that we don't need the first sample run (`-d cpu`). The second one (`-d cuda`) is good to keep I think -- as we move more code to the gpu, it will become faster and we can keep checking it for correctness through the autotests.

Reply:
`done`